flang/include/flang/Optimizer/Builder/HLFIRTools.h

```diff
@@ -243,7 +243,8 @@ hlfir::AssociateOp genAssociateExpr(mlir::Location loc,
                                     fir::FirOpBuilder &builder,
                                     hlfir::Entity value,
                                     mlir::Type variableType,
-                                    llvm::StringRef name);
+                                    llvm::StringRef name,
+                                    std::optional<mlir::NamedAttribute> attr);
```

Comment:
nit to avoid code churn
```suggestion
                                    std::optional<mlir::NamedAttribute> attr = std::nullopt);
```